java/src/jmri/jmrix/can/cbus/swing/nodeconfig/CbusAllocateNodeNumber.java

```diff
@@ -270,11 +281,16 @@ private void processNodeParams(CanReply m) {
         nodepropbuilder.append( CbusNodeConstants.getModuleType( _paramsArr[0] , _paramsArr[2] ));
             
         if (WAITINGRESPONSE_RQNN_PARAMS) {
-            rqNNtext.setText(nodepropbuilder.toString());
+            ThreadingUtil.runOnGUI( ()->{
+                rqNNtext.setText(nodepropbuilder.toString());
+            });
+            
             WAITINGRESPONSE_RQNN_PARAMS=false;
         }
         else if (!NODE_NUM_DIALOGUE_OPEN) {
-            startnodeallocation( -1, nodepropbuilder.toString() );
+            ThreadingUtil.runOnGUI( ()->{
```

Comment:
I would think if anything you want to run this on the LayoutThread, not the Gui thread (these may currently be the same, but they don't have to be long term...)

Reply:
The jast majority of startnodeallocation is displaying, then waiting from a response from a javax.swing.JOptionPane